Sources/IRGen/Conformance/IRConformanceProcessor.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  IRConformanceProcessor.swift
+//  IRGen
+//
+//  Created by Niklas Vangerow on 20/10/2018.
+//
+
+import AST
+import Lexer
+
+/// A prepocessing step to update the program's AST before code generation, specifically in order to resolve Self
+
+public struct IRConformanceProcessor: ASTPass {
```

Comment:
Unsure about the name of this pass and the comment above.

About the name – it is not related to the IR very much, since it deals with traits and `Self`.

About the comment - this AST pass actually also copies over defaulted functions from traits. This should be mentioned.

`ConformanceProcessor`? `ConformanceResolver`? `TraitResolver`?

Reply:
TraitResolver sounds good to me. Good points, will address with the comment and stuff